### DISCUSSION TITLE
Can files in vault get sync without opening obsidian?

### DISCUSSION THREAD

**Discussion by qlibp** [vrtmrz/obsidian-livesync#421] [Q&A] [answered]
For example, I edit files in vault using vim, can live sync server automatically detect it and perform the sync?

**Comment by SystemZ** [comment]
I have similar question, can I append lines by curl or file editing?
I'm trying to integrate home assistant to write to my notes but I'm worried about file conflicts.

**Reply by qlibp** [reply, marked as answer]
Should work. I try with my macOS and iOS. On macos, I open obsidian in one window and I edit it using some other editors, once I save the file, the info get synced on my phone.

**Reply by vrtmrz** [reply]
We can modify the file that is stored in the remote CouchDB by CURL request if the remote database is not encrypted.
If the remote is encrypted, we have to use some items. A bit outdated, but the [DirectFileManipulator.ts](https://github.com/vrtmrz/livesync-commonlib/blob/main/src/API/DirectFileManipulator.ts) is available.

**Reply by SystemZ** [reply]
@vrtmrz Thx for reply! Can you show the simplest example for curl with unencrypted DB? File you provided is pretty complicated for non-JS folk like me and it's hard to start 😅

**Comment by rewse** [comment]
One additional condition is that the edited file does not have to be open, but an Obsidian window must be open.

**Comment by alx-xlx** [comment]
I use [obsidian docker](https://docs.linuxserver.io/images/docker-obsidian/) to do this. I think this will keep the obsidian instance running in background at all times in some sort of way using the kasmvnc.

I don't know about the efficiency of the cpu, ram usage in this. But it will work for you, as you don't need to manually start anything.

**Reply by qlibp** [reply]
interesting, I think it should work

**Reply by slimjet4u** [reply]
the livesync plugin takes too much resources while synchronizing
